client/client-multi/client-cpp/test/impl/utils/ThreadPoolTest.cpp

```diff
@@ -248,54 +248,82 @@ BOOST_AUTO_TEST_CASE(AwaitTerminationAllTaskCompletedTest)
 
 BOOST_AUTO_TEST_CASE(AwaitTerminationTimeoutTest)
 {
-    std::mutex mutex;
-    std::condition_variable onStartCondition;
 
-    ThreadPool threadPool;
+    class Condition final
+    {
+    public:
 
-    bool isStart = false;
-    std::size_t taskExecutionTime = 1;
-    std::uint16_t expectedTaskCount = 0;
-    std::atomic_uint actualTaskCount(0);
-    std::size_t timeToWaitAllTasks = 0;
+        explicit Condition(const bool state = false) :
+        mutex_(), variable_(), state_(state)
+        {}
 
-    ThreadPoolTask task = [&mutex, &onStartCondition, &actualTaskCount, &isStart, &taskExecutionTime] ()
-                              {
-                                    {
-                                        std::unique_lock<std::mutex> lock(mutex);
-                                        onStartCondition.wait(lock, [&isStart] () { return isStart; });
-                                    }
-                                    std::this_thread::sleep_for(std::chrono::seconds(taskExecutionTime));
-                                    actualTaskCount++;
-                              };
+        bool state()
+        {
+            std::unique_lock<std::mutex> lock(mutex_);
+            return state_;
+        }
 
-    // Executed.
-    threadPool.add(task);
-    ++expectedTaskCount;
-    timeToWaitAllTasks += taskExecutionTime;
+        bool state(const bool newState)
+        {
+            std::unique_lock<std::mutex> lock(mutex_);
 
-    // Executed.
-    threadPool.add(task);
-    ++expectedTaskCount;
-    timeToWaitAllTasks += taskExecutionTime;
+            if(state_ == newState)
```

Comment:
Please, follow the Code Style guide:
http://kaaproject.github.io/kaa/docs/v0.10.0/Customization-guide/Code-style/C-C++/